code-challenges.test.js

```diff
@@ -21,10 +21,45 @@ const secretCodeWord2 = "Gobbledygook"
 const secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
 
+// describe("makeCodedMessage", () => {
+//     it("returns coded message", () => {
+//         expect(makeCodedMessage(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+//         expect(makeCodedMessage(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+//         expect(makeCodedMessage(secretCodeWord3)).toEqual("3cc3ntr1c")
+//     })
+// })
+
 // b) Create the function that makes the test pass.
 
+const makeCodedMessage = (str) => {
+    var stringArr = str.toLowerCase().split("")
+    for(let i=0; i<stringArr.length; i++){
+        if(stringArr[i] === 'a') {stringArr[i] = '4'}
+        if(stringArr[i] === 'e') {stringArr[i] = '3'}
+        if(stringArr[i] === 'i') {stringArr[i] = '1'}
+        if(stringArr[i] === 'o') {stringArr[i] = '0'}
+    }
+    return  stringArr.join("").toUpperCase()[0] + stringArr.slice(1).join("")
+}
+
+// console.log(makeCodedMessage(secretCodeWord1))
+// console.log(makeCodedMessage(secretCodeWord2))
+// console.log(makeCodedMessage(secretCodeWord3))
+
 // Pseudo code:
 
+// crete makeCodedMessage(str) => {
+//     Convert str to lowercase and split it into an array of characters called stringArr
+//     For each character c in stringArr
+//         If c is 'a', replace it with '4'
+//         If c is 'e', replace it with '3'
+//         If c is 'i', replace it with '1'
+//         If c is 'o', replace it with '0'
+//     Return join() stringArr (converting to string) and then index into zeroth place to capitalize first character in word
+//     and then add that to to rest of of stringArr starting at index 1.
```

Comment:
Good logic, I suggest shortening this pseudo code on lines 54-57, as it's basically repeating your conditional in the code.